Extra/Shaders/NewVegasReloaded/Shaders/ISHDRBLENDINSHADERCINAM.pso.hlsl

```diff
@@ -51,37 +51,33 @@ struct VS_OUTPUT {
+    float gammaCorrection = max(1.0, TESR_ToneMapping.w);
 
+    float4 background = tex2D(DestBlend, IN.ScreenOffset.xy); // sdr image (already tonemapped) displayed within the mask
+    float4 bloom = tex2D(Src0, IN.ScreenOffset.xy);
+    float4 final = tex2D(DestBlend, IN.texcoord_1.xy);
+    bloom.rgb = pows(bloom.rgb, max(1.0, gammaCorrection)); // linearize bloom
+    final.rgb = pows(final.rgb, max(1.0, gammaCorrection)); // linearize color
+    
     // scale bloom while maintaining color
-    float4 bloom = tex2D(ScreenSpace, IN.ScreenOffset.xy);
-    //float maxBloom = max(bloom.r, max(bloom.g, bloom.b));
-    // float4 ratio = bloom / maxBloom;
-    // bloom.rgb = TESR_HDRBloomData.x * pows(max(maxBloom, 0), TESR_HDRBloomData.y) * ratio;
     bloom.rgb = TESR_HDRBloomData.x * pows(max(bloom.rgb, 0.0), TESR_HDRBloomData.y);
 
-    float gammaCorrection = max(1.0, TESR_ToneMapping.w);
-    float4 hdrImage = tex2D(DestBlend, IN.texcoord_1.xy);
+    float cinematicScalar = TESR_HDRData.z;
+    final.rgb = lerp(luma(final.rgb).xxx, final.rgb, Cinematic.x * cinematicScalar); // saturation
+    final.rgb = lerp(final.rgb, Tint.rgb * luma(final.rgb), saturate(Tint.a * TESR_ToneMapping.z)); // apply tint
     
-    float3 final = pows(hdrImage.rgb, gammaCorrection); // linearize
-
-    float screenluma = luma(final);
-    final = lerp(screenluma.xxx, final, Cinematic.x * TESR_HDRData.z); // saturation
-    final = lerp(final.rgb, Tint.rgb * luma(final.rgb), saturate(Tint.a * TESR_ToneMapping.z)); // apply tint
-    
-    float q0 = 1.0 / max(bloom.w, HDRParam.x); // HDRParam.x, brights cutoff?
-    final = ((q0 * HDRParam.x) * final) + max(bloom.rgb * (q0 * 0.5), 0.0); // blend image and bloom
+    float q0 = 1.0 / max(bloom.w, HDRParam.x); // HDRParam.x, brights cutoff
+    final.rgb = ((q0 * HDRParam.x) * final.rgb) + max(bloom.rgb * (q0 * (TESR_DebugVar.x * 0.5)), 0.0); // blend image and bloom
     
-    final = lerp(final.rgb, final.rgb * Cinematic.z, TESR_HDRData.z); // apply brightness from Cinematic
-    final = tonemap(final.rgb * TESR_HDRData.y); // exposure & tonemap using provided tonemapper
+    final.rgb = lerp(final.rgb, final.rgb * Cinematic.z, cinematicScalar); // apply brightness from Cinematic
+    final.rgb = tonemap(final.rgb * TESR_HDRData.y); // exposure & tonemap using provided tonemapper
     
-    final = pows(final.rgb, 1.0/max(1.0, TESR_HDRData.w)); // delinearise
+    final.rgb = pows(final.rgb, 1.0/max(1.0, TESR_HDRData.w)); // delinearise
     
-    final = lerp(final.rgb, Cinematic.z * (final.rgb - Cinematic.y) + Cinematic.y, TESR_HDRData.z * TESR_ToneMapping.y); // apply contrast from Cinematic, scaled by modifier
-    final = lerp(final.rgb, Fade.rgb, Fade.a); // apply night eye and fade, gamma-space but vanilla accurate
+    final.rgb = lerp(final.rgb, Cinematic.w * (final.rgb - Cinematic.y) + Cinematic.y, cinematicScalar * TESR_ToneMapping.y); // apply contrast from Cinematic, scaled by modifier
```

Comment:
same question here as https://github.com/mchaptel/TESReloaded10/pull/33/files#diff-ae7d9f078a232fd5a48ee69ba0b5e62e6146b76ba65caced9b07045f875e8641R74

Reply:
Same mistake, also fixed. .z was also being used as exposure, which is now .w